js/ui/appDisplay.js

```diff
@@ -2807,7 +2810,50 @@ class HackAppIcon extends AppIcon {
                 DownstreamEndlessIconState.HACK_DEACTIVATED;
         });
 
+        this._hackPulseId = global.settings.connect('changed::hack-icon-pulse', () => {
+            this._pulse = global.settings.get_boolean('hack-icon-pulse');
+            if (this._pulseWaitId) {
+                Mainloop.source_remove(this._pulseWaitId);
+                this._pulseWaitId = 0;
+            }
+            this.pulse();
+        });
+
         this.canDrop = false;
+
+        this.pulse();
+    }
+
+    pulse() {
+        if (!this._pulse)
+            return;
+
+        const params = { time: 0.1, transition: 'easeOutQuad' };
+        this._tweenIcon({...params, scale_x: 1.1, scale_y: 1.1})
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 0.9, scale_y: 0.9}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 1.1, scale_y: 1.1}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 0.9, scale_y: 0.9}))
+            .then(this._tweenIcon.bind(this, {...params, scale_x: 1, scale_y: 1}))
```

Comment:
This is a bit different from the rest of the code base (though, it's an interesting approach!) I wonder if using promises won't conflict with the GLib mainloop.

Anyway, a minor nitpick: `1.0` instead of `1`.

Reply:
I think that there's no problem with Promises and mainloop, but maybe @ptomato will know better.

Reply:
There should be no conflict, a promise's callback will be called in the next idle iteration after it resolves.